Vdo.ai

cpuBird opened the pull request:
<!--
# Instructions:

- Pick a meaningful title for your pull request. (Use sentence case.)
  - Prefix the title with an emoji to identify what is being done. (Copy-paste the emoji from the list below.)
  - Do not overuse punctuation in the title (like `(chore):`).
  - If it is helpful, use a simple prefix (like `ProjectX: Implement some feature`).
- Enter a succinct description that says why the PR is necessary, and what it does.
  - Mention the GitHub issue that is being addressed by the pull request.
  - The keywords `Fixes`, `Closes`, or `Resolves` followed the issue number will automatically close the issue.

> NOTE: All non-trivial changes (like introducing new features or components) should have an associated issue or reference an I2I (intent-to-implement: go.amp.dev/i2i). Please read through the contribution process (go.amp.dev/contributing/code) for more information.

# Example of a good description:

- Implement aspect X
- Leave out feature Y because of A
- Improve performance by B
- Improve accessibility by C

# Emojis for categorizing pull requests (copy-paste emoji into description):

✨ New feature
🐛 Bug fix
🔥 P0 fix
✅ Tests
🚀 Performance improvements
🖍 CSS / Styling
♿ Accessibility
🌐 Internationalization
📖 Documentation
🏗 Infrastructure / Tooling / Builds / CI
⏪ Reverting a previous change
♻️ Refactoring
🚮 Deleting code
-->
VDO.AI : Implemented VDO.AI in amp-ad 
✨ New feature

zhouyx commented:
to @ampproject/wg-ads

powerivq commented on ads/vdo.js:
```diff
@@ -0,0 +1,47 @@
+ */
+export function vdo(global, data) {
+  /*eslint "google-camelcase/google-camelcase": 0*/
+  console.log(global, data);
```
Do you need this for production?

cpuBird replied:
Thanks for identifying this piece of code. We would like to remove this for production purposes. This was being used by us during development stage. Can you please remove this from your end? That would be great :D

cpuBird replied:
I have updated the code myself. No need for further modifications. Please have a final look. Thanks !

powerivq commented:
@cpuBird Hi. I approved the PR, but
- CLA needs to be resolved. See comments above.
- Lint is failing. Please fix the issue.

cpuBird commented:
@googlebot I signed it!

cpuBird commented:
Hi @powerivq,
Thanks for approving the PR. 
1. I've signed the CLA.
2. The linting issue mentioned above was because of the CRLF line endings in windows. I've changed that to LF and the issue was resolved. I've commited the code pls have a look and tell me if anything required. :)

powerivq commented:
> Hi @powerivq,
> Thanks for approving the PR.
> 
> 1. I've signed the CLA.
> 2. The linting issue mentioned above was because of the CRLF line endings in windows. I've changed that to LF and the issue was resolved. I've commited the code pls have a look and tell me if anything required. :)

Not sure why but the bot did not pick up your CLA. Can you respond "I signed it" again? Also, the build issue is still there.

cpuBird commented:
> > Hi @powerivq,
> > Thanks for approving the PR.
> > 
> > 1. I've signed the CLA.
> > 2. The linting issue mentioned above was because of the CRLF line endings in windows. I've changed that to LF and the issue was resolved. I've commited the code pls have a look and tell me if anything required. :)
> 
> Not sure why but the bot did not pick up your CLA. Can you respond "I signed it" again? Also, the build issue is still there.

Hi @powerivq ,
CLA is accepted and I've also fixed the linting issues

powerivq commented:
@cpuBird plz ping me when it passes. I see it still failing.

cpuBird commented:
Hi @powerivq , 
Sorry for delay in revert. You told me that I have linting issues in commets above so I fixed them using the following commands mentioned on - https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md : 
`gulp lint --local_changes --fix` &
`gulp prettify --local_changes --fix`

I would appreciate if you can help me figure the exact issue so that i can fix it.

powerivq commented:
@cpuBird You can click on the travis build result https://travis-ci.org/ampproject/amphtml/jobs/629961620?utm_medium=notification&utm_source=github_status

```
Type checking failed:
ads/vdo.js:26: ERROR - [JSC_POSSIBLE_INEXISTENT_PROPERTY] Property unitid never defined on Object
    unitData: data.unitid,
                   ^^^^^^
ads/vdo.js:35: ERROR - [JSC_POSSIBLE_INEXISTENT_PROPERTY] Property tagname never defined on Object
    ADFRAME_URL: 'https://a.vdo.ai/core/' + data.tagname + '/adframe.js' };
                                                 ^^^^^^^
```

cpuBird commented:
Hi @powerivq ,
Please have a look. I've fixed the issues with failing build.

powerivq commented:
@cpuBird You can look at the build status badge below. Everything except `owners-check` should succeed (showing a tick rather than X). You can check the `Details` button on `continuous-integration/travis-ci/pr — The Travis CI build failed` to see what went wrong. The local and remote test failures are related to your changes. The saucelabs seems to be a flaky test that would succeed.

cpuBird commented:
@powerivq Done

honeybadgerdontcare commented:
Drive by comment: if you're adding new files please have the copyright be the current year. In this instance it may be the first `2020` copyright: yeah!

cpuBird commented:
Both done.
